zkevm-circuits/src/evm_circuit/step.rs

```diff
@@ -0,0 +1,553 @@
+use crate::{
+    evm_circuit::{
+        execution::bus_mapping_tmp::{Call, ExecStep},
+        param::{STEP_HEIGHT, STEP_WIDTH},
+        util::Cell,
+    },
+    util::Expr,
+};
+use bus_mapping::evm::OpcodeId;
+use halo2::{
+    arithmetic::FieldExt,
+    circuit::Region,
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression},
+};
+use std::collections::VecDeque;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum ExecutionResult {
```

Comment:
`ExecutionState` sounds much better since it's acting as a finite state machine.

Reply:
Fixed in [`426786e`](https://github.com/appliedzkp/zkevm-circuits/pull/196/commits/426786e23c0eb5d69ca65df86fe67d91b751e1d9)

Reply:
Isn't it corresponding to `ExecutionResult` in `go-ethereum/internal/ethapi/api.go`. If that's the case, we should keep the naming consistent with **geth**?

Reply:
I think the enum here is more like **specify which kind of constraints the step enables** instead of **`ExecutionResult` of a call** in geth.

Currently EVM circuit is built to be like a finite state machine, each step verifies a opcode in most time (except multi-step opcode or precompiled), and the `ExecutionState` here specified which kind of constraint to enable to verify the step.

Not sure if it makes sense, any further suggestions are appreciated.